prow/cmd/deck/pr_history.go

```diff
@@ -268,8 +270,19 @@ func getPRHistory(url *url.URL, config *config.Config, gcsClient *storage.Client
 	// job name -> commit hash -> list of builds
 	jobCommitBuilds := make(map[string]map[string][]buildData)
 
-	for bucketName, gcsPaths := range toSearch {
-		bucket := gcsBucket{bucketName, gcsClient.Bucket(bucketName)}
+	for bucket, gcsPaths := range toSearch {
+		// bucket is the bucket field of the GCSConfiguration, which means it could be missing the
+		// storageType prefix (but it's deprecated to use a bucket name without <storage-type>:// prefix)
+		parsedBucket, err := url.Parse(bucket)
+		if err != nil {
+			return template, fmt.Errorf("parse bucket %s: %v", bucket, err)
```

Comment:
Really would prefer we use %w to %v here. It is useful to include the inner error in case the caller has a need to consider it.

%w should be our reflexive behavior for `return fmt.Errorf()` and the only reason it isn't is because this feature is very new.

Reply:
Yup got it. Some of the code was written before the last review and probably I also forgot it. Will try to default to it in the future

Reply:
Done also changed it in a few other cases where I touched the code.